descwl_shear_testing/mbobs_extractor.py

```diff
@@ -0,0 +1,437 @@
+"""
+copied from meas_extensions_ngmix and modified
+"""
+import numpy as np
+import ngmix
+import lsst.log
+import lsst.afw.image as afw_image
+from lsst.pex.exceptions import InvalidParameterError, LogicError
+import lsst.geom as geom
+
+from . import util
+
+
+class MBObsMissingDataError(Exception):
+    """
+    Some number was out of range
+    """
+
+    def __init__(self, value):
+        super(MBObsMissingDataError, self).__init__(value)
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+
+class MBObsExtractor(object):
+    """
+    class to extract observations from the images
+
+    parameters
+    ----------
+    config: dict or dict like
+        A dictionary configuration
+    exposures: list of exposures
+        List of ExposureF
+    sources: records
+        records from detection
+    psfs: optional
+        List of psf images
+    """
+
+    def __init__(self, *, config, exposures, sources, psfs=None):
+        self.config = config
+        self.exposures = exposures
+        self.sources = sources
+        self.psfs = psfs
+
+        self.log = lsst.log.Log.getLogger("MBObsExtractor")
+
+        self._verify()
+
+    def get_mbobs_list(self, weight_type='weight'):
+        """
+        Get a list of `MultiBandObsList` for all or a set of objects.
+
+        Parameters
+        ----------
+        weight_type: string, optional
+            Currently only 'weight' is supported
+
+        Returns
+        -------
+        mbobs_list : list of ngmix.MultiBandObsList
+            The list of `MultiBandObsList`s for the requested objects.
+        """
+
+        list_of_obs = []
+
+        for rec in self.sources:
+            mbobs = self._get_mbobs(rec, weight_type=weight_type)
+            list_of_obs.append(mbobs)
+
+        return list_of_obs
+
+    def _get_mbobs(self, rec, weight_type='weight'):
+        """
+        make an ngmix.MultiBandObsList for input to the fitter
+
+        parameters
+        ----------
+        images: dict
+            A dictionary of image objects
+        rec: object record
+            TODO I don't actually know what class this is
```

Comment:
:)

Reply:
yeah, there are a few of these that I don't know yet